pkg/catalog/broadcaster.go

```diff
@@ -0,0 +1,101 @@
+package catalog
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/openservicemesh/osm/pkg/configurator"
+)
+
+type Broadcaster struct {
```

Comment:
`pkg/catalog/repeater.go` does the broadcasting right now. Why isn't that reusable?

Reply:
since we want this to be a dynamic toggle that can be changed at runtime, repeater will not work since the ticker is created as a static object once and never refreshed. 

I added some content to the ticket here - https://github.com/openservicemesh/osm/issues/1501#issuecomment-673858429

Reply:
Can't you augment repeater with the dynamic toggle?

Reply:
I suppose the same logic could be made part of the repeater and be a set of methods associated with `MeshCatalog` but it wouldn't change the functionality right? Unless you have another idea of augmenting it?

Reply:
I feel augmenting the repeater code provides context on how the code evolved, yes it wouldn't differ in functionality to what you are attempting to do. You probably want to remove repeater completely if its not going to be used and feel its not extendable.

Reply:
Makes sense. I am still not able to make it work in this WIP PR. The broadcaster fails to send announcements and am having trouble triaging what the problem is 😕 . But I will look into combining the two

Reply:
Wouldn't it be easier to pass in `config.Configurator` to repeater.go and have it use the Configurator interface to read the ticker config?

Reply:
`repeater.go` > `repeater()` already has access to `config.Configurator` through it's associated `MeshCatalog` object
```go
func (mc *MeshCatalog) repeater() {
...

type MeshCatalog struct {
	endpointsProviders []endpoint.Provider
	meshSpec           smi.MeshSpec
	...
	configurator       configurator.Configurator
        ...
	announcementChannels mapset.Set
}
```

But yea if we add a broadcaster loop in repeater.go then it would serve the same functionality (I suppose)

Reply:
@shashankram Just to clarify, do you want me to get rid of `broadcaster.go` and also of `Broadcaster` struct and consolidate everything in `MeshCatalog`? For the solution, I will need a bunch of member fields, like locks, channels, etc. Is that what you were imagining?

Reply:
Yes, the broadcast functionality should ideally be augmented within the catalog as a part of `repeater.go`.